packages/navigator/src/browser/navigator-widget.tsx

```diff
@@ -88,21 +92,15 @@ export class FileNavigatorWidget extends FileTreeWidget {
                         this.model.expandNode(child);
                     }
                 }
+            }),
+            this.workspaceService.onWorkspaceChanged(async event => {
```

Comment:
It can be moved to `NavigatorModel` since it already knows about `WorkspaceService`, as well as calling `updateRoot` on the initialization.